examples/cpp/dds/CustomPayloadPoolExample/CustomPayloadPool_main.cpp

```diff
@@ -0,0 +1,237 @@
+#include <fastrtps/log/Log.h>
+
+#include <optionparser.hpp>
+
+using eprosima::fastdds::dds::Log;
+
+namespace option = eprosima::option;
+
+struct Arg : public option::Arg
+{
+    static void print_error(
+            const char* msg1,
+            const option::Option& opt,
+            const char* msg2)
+    {
+        fprintf(stderr, "%s", msg1);
+        fwrite(opt.name, opt.namelen, 1, stderr);
+        fprintf(stderr, "%s", msg2);
+    }
+
+    static option::ArgStatus Unknown(
+            const option::Option& option,
+            bool msg)
+    {
+        if (msg)
+        {
+            print_error("Unknown option '", option, "'\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Required(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0 && option.arg[0] != 0)
+        {
+            return option::ARG_OK;
+        }
+
+        if (msg)
+        {
+            print_error("Option '", option, "' requires an argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus String(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            return option::ARG_OK;
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires an argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+};
+
+enum  optionIndex
+{
+    UNKNOWN_OPT,
+    HELP
+};
+
+const option::Descriptor usage[] = {
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,
+      "Usage: CustomPayloadPoolExample <publisher|subscriber>\n\nGeneral options:" },
+    { HELP,    0, "h", "help",               Arg::None,      "  -h \t--help  \tProduce help message." },
+    {0, 0, 0, 0, 0, 0}
+};
+
+int main(
+        int argc,
+        char** argv)
+{
+    int columns;
+
+#if defined(_WIN32)
+    char* buf = nullptr;
+    size_t sz = 0;
+    if (_dupenv_s(&buf, &sz, "COLUMNS") == 0 && buf != nullptr)
+    {
+        columns = strtol(buf, nullptr, 10);
+        free(buf);
+    }
+    else
+    {
+        columns = 80;
+    }
+#else
+    columns = getenv("COLUMNS") ? atoi(getenv("COLUMNS")) : 80;
+#endif // if defined(_WIN32)
+
+    std::cout << "Starting " << std::endl;
+
+    int type = 1;
+    uint32_t count = 10;
+    uint32_t sleep = 100;
+
+    argc -= (argc > 0);
+    argv += (argc > 0); // skip program name argv[0] if present
+    option::Stats stats(true, usage, argc, argv);
+    std::vector<option::Option> options(stats.options_max);
+    std::vector<option::Option> buffer(stats.buffer_max);
+    option::Parser parse(true, usage, argc, argv, &options[0], &buffer[0]);
+
+    try
+    {
+        if (parse.error())
+        {
+            throw 1;
+        }
+
+        if (options[HELP] || options[UNKNOWN_OPT])
+        {
+            throw 1;
+        }
+
+        if (parse.nonOptionsCount() != 1)
```

Comment:
Else I cannot configure count not sleep for the publisher, which BTW are not in the help message.
```suggestion
        if (parse.nonOptionsCount() <1)
```